src/Cake.Bakery/Scripting/PackageReferenceComparer.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using Cake.Core.Packaging;
+
+namespace Cake.Bakery.Scripting
+{
+    internal sealed class PackageReferenceComparer : IEqualityComparer<PackageReference>
+    {
+        public bool Equals(PackageReference x, PackageReference y)
+        {
+            if (x == null && y == null)
+            {
+                return true;
+            }
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return x.OriginalString.Equals(y.OriginalString, StringComparison.OrdinalIgnoreCase);
```

Comment:
`x.OriginalString` could theoretically be null here. Change to `string.Equals(x,y,comparison)`.